core/src/main/java/org/infinispan/commands/read/EntrySetCommand.java

```diff
@@ -106,8 +105,7 @@ public String toString() {
             return Closeables.iterator(dataContainer.iterator());
          }
          Iterator<CacheEntry<K, V>> iterator = new DataContainerRemoveIterator<>(cache, dataContainer);
-         RemovableIterator<CacheEntry<K, V>> removableIterator = new RemovableIterator<>(iterator, e -> cache.remove(e.getKey(), e.getValue()));
-         return new IteratorMapper<>(removableIterator, e -> new EntryWrapper<>(cache, e));
+         return new IteratorMapper<>(iterator, e -> new EntryWrapper<>(cache, e));
```

Comment:
I sense an opportunity to eliminate some more lambdas :) There are 3 more instances of `e -> new EntryWrapper<>(cache, e)`, so it would be better to write a new iterator that handles both `Iterator.remove()` and `Entry.setValue()`.

Ideally though, I'd like us to stop wrapping `EntrySet` in the interceptor chain: replace `EntrySetCommand`/`KeySetCommand`/`SizeCommand` with a single `IterateCommand` that returns a read-only `Iterator` /`Spliterator`/`Publisher` (whatever is easier to implement), and only wrap it into an entries/keys/values collection once, in `CacheImpl` (removing `DistributedBulkInterceptor` would also be a plus).

Off-topic, since the main topic is lambdas now, `DataContainerRemoveIterator` is too specific, it could use any `Iterator`.